src/extension/infer.rs

```diff
@@ -279,7 +279,13 @@ impl UnificationContext {
                     let m_input_node = self.make_or_get_meta(input, dir);
                     self.add_constraint(m_input_node, Constraint::Equal(m_input));
                     let m_output_node = self.make_or_get_meta(output, dir);
-                    self.add_constraint(m_output_node, Constraint::Equal(m_output));
+                    // If the parent node is a FuncDefn, it's input_resources
```

Comment:
input_resources -> input_extensions

Reply:
in fact, "it's input resources" -> "its input_extensions" (it's = it is, its = belonging to it)